src/chainstate/stacks/miner.rs

```diff
@@ -3994,6 +4006,131 @@ pub mod test {
+            // send transactions to the mempool
+            let tip = BurnDB::get_canonical_burn_chain_tip(&peer.burndb.as_ref().unwrap().conn()).unwrap();
+
+            let (burn_ops, stacks_block, microblocks) = peer.make_tenure(|ref mut miner, ref mut burndb, ref mut chainstate, vrf_proof, ref parent_opt, ref parent_microblock_header_opt| {
+                let parent_tip = match parent_opt {
+                    None => {
+                        StacksChainState::get_genesis_header_info(&chainstate.headers_db).unwrap()
+                    }
+                    Some(block) => {
+                        let ic = burndb.index_conn();
+                        let snapshot = BurnDB::get_block_snapshot_for_winning_stacks_block(&ic, &tip.burn_header_hash, &block.block_hash()).unwrap().unwrap();      // succeeds because we don't fork
+                        StacksChainState::get_anchored_block_header_info(&chainstate.headers_db, &snapshot.burn_header_hash, &snapshot.winning_stacks_block_hash).unwrap().unwrap()
+                    }
+                };
+
+                let parent_header_hash = parent_tip.anchored_header.block_hash();
+                let parent_tip_bhh = parent_tip.burn_header_hash.clone();
+                let coinbase_tx = make_coinbase(miner, tenure_id);
+
+                let mut mempool = MemPoolDB::open(false, 0x80000000, &chainstate_path).unwrap();
+
+                if tenure_id == 2 {
+                    let contract = "
+                    (define-data-var bar int 0)
+                    (define-public (get-bar) (ok (var-get bar)))
+                    (define-public (set-bar (x int) (y int))
+                      (begin (var-set bar (/ x y)) (ok (var-get bar))))";
+
+                    // should be mined once
+                    let contract_tx = make_user_contract_publish(&privks[tenure_id], 0, 10000, &format!("hello-world-{}", tenure_id), &contract);
+                    let mut contract_tx_bytes = vec![];
+                    contract_tx.consensus_serialize(&mut contract_tx_bytes).unwrap();
+                    mempool.submit_raw(&parent_tip_bhh, &parent_header_hash, contract_tx_bytes).unwrap();
+
+                    eprintln!("first tx submitted");
+                    // eprintln!("\n\ntransaction:\n{:#?}\n\n", &contract_tx);
+
+                    sleep_ms(2000);
+
+                    // should never be mined
+                    let contract_tx = make_user_contract_publish(&privks[tenure_id], 1, 10000, &format!("hello-world-{}-2", tenure_id), &contract);
+                    let mut contract_tx_bytes = vec![];
+                    contract_tx.consensus_serialize(&mut contract_tx_bytes).unwrap();
+                    mempool.submit_raw(&parent_tip_bhh, &parent_header_hash, contract_tx_bytes).unwrap();
+
+                    eprintln!("second tx submitted");
+                    // eprintln!("\n\ntransaction:\n{:#?}\n\n", &contract_tx);
+
+                    sleep_ms(2000);
```

Comment:
Before merging this PR, please remove the test's calls to `sleep_ms()`.

Reply:
I just copied from the test above, which was using the `sleep_ms()`.   I'll remove if not needed.